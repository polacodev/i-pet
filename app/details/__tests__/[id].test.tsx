import { render, screen } from '@testing-library/react-native';
import '@testing-library/react-native/extend-expect';
import React from 'react';
import { Alert } from 'react-native';

import PetPage from '@/app/details/[id]';

const petObj = {
  id: '00000000-0000-0000-0000-000000000000',
  inserted_at: 'testInsertedId',
  pet_image: 'https://fakeurl/fakeimage/test.png',
  pet_age: 'testPetAge',
  pet_breed: 'testPetBreed',
  pet_gender: 'testPetGender',
  pet_medical_condition: 'testPetMC',
  pet_name: 'testPetName',
  pet_type: 'testPetType',
  profiles: {
    country_code: 'testProfileCountryCode',
    email: 'testProfileEmail',
    id: 'testProfileId',
    phone: 'testProfilePhone',
  },
  user_id: 'testUserId',
};

const petObjInvalid =
  '{"id":"00000000-0000-0000-0000-000000000000","inserted_at":testInse rtedId,"pet_image":"https://fakeurl/fakeimage/test.png","pet_age":"testPetAge","pet_breed":"testPetBreed","pet_gender":"testPetGender","pet_medical_condition":"testPetMC","pet_name":"testPetName","pet_type":"testPetType","profiles":{"country_code":"testProfileCountryCode","email":"testProfileEmail","id":"testProfileId","phone":"testProfilePhone"},"user_id":"testUserId"}';

jest.mock('expo-router', () => ({
  useGlobalSearchParams: jest.fn(),
}));

jest.mock('@/localizations/localization', () => ({
  localization: {
    t: jest.fn().mockImplementation((key) => key),
  },
}));

jest.mock('@/localizations/localization', () => ({
  localization: {
    t: jest.fn((key) => {
      const translations = {
        id_error_alert_title: 'Error',
        id_error_alert_message: 'Error Message',
      };
      return translations[key] || key;
    }),
  },
}));

describe('<PetPage />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  test('PetPage renders the PetCard with valid petData and Snapshot', () => {
    require('expo-router').useGlobalSearchParams.mockReturnValue({
      petData: JSON.stringify(petObj),
    });
    const { toJSON } = render(<PetPage />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('PetPage renders the PetCard with invalid petData and Snapshot', () => {
    require('expo-router').useGlobalSearchParams.mockReturnValue({
      petData: petObjInvalid,
    });
    const parseAlertError = jest.spyOn(Alert, 'alert');
    render(<PetPage />);
    expect(parseAlertError).toHaveBeenCalled();
  });

  test('PetPage fails the JSON parse and shows up the alert and Snapshot', () => {
    require('expo-router').useGlobalSearchParams.mockReturnValue({
      petData: null,
    });
    render(<PetPage />);
    const textForm = screen.getAllByTestId('pet-text');
    expect(screen.toJSON()).toMatchSnapshot();
    expect(textForm[0]).toHaveTextContent(' : ');
  });
});
