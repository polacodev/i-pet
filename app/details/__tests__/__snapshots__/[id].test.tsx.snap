// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PetPage /> PetPage fails the JSON parse and shows up the alert and Snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f6f6f6",
      },
      {
        "flex": 1,
        "justifyContent": "flex-start",
        "width": "100%",
      },
    ]
  }
  testID="pet-view"
>
  <View
    style={
      [
        {
          "backgroundColor": "#f6f6f6",
        },
        {
          "alignItems": "center",
        },
      ]
    }
    testID="pet-view"
  >
    <View
      style={
        [
          {
            "backgroundColor": "#f6f6f6",
          },
          {
            "alignItems": "center",
          },
        ]
      }
      testID="pet-view"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#ffffff",
              "fontSize": 80,
            },
            [
              undefined,
            ],
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="pet-icon"
      >
        
      </Text>
    </View>
    <Text
      style={
        [
          {
            "color": "#155e75",
          },
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "lineHeight": 28,
          },
          {
            "paddingTop": 8,
          },
        ]
      }
      testID="pet-title"
    />
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      : 
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      : 
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      : 
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      : 
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      : 
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      : 
    </Text>
    <Text
      style={
        [
          {
            "color": "#155e75",
          },
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "lineHeight": 28,
          },
          {
            "paddingTop": 8,
          },
        ]
      }
      testID="pet-title"
    />
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      : 
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      undefined:  
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#075E54",
          "borderRadius": 10,
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
          "margin": 6,
          "opacity": 1,
          "padding": 8,
          "width": "85%",
        }
      }
      testID="pet-button"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#ffffff",
              "fontSize": 30,
            },
            [
              undefined,
            ],
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="pet-icon"
      >
        
      </Text>
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#ffffff",
            },
          ]
        }
        testID="pet-text"
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0e7490",
          "borderRadius": 10,
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
          "margin": 6,
          "opacity": 1,
          "padding": 8,
          "width": "85%",
        }
      }
      testID="pet-button"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#ffffff",
              "fontSize": 30,
            },
            [
              undefined,
            ],
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="pet-icon"
      >
        
      </Text>
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#ffffff",
            },
          ]
        }
        testID="pet-text"
      >
        QR Code
      </Text>
    </View>
  </View>
</View>
`;

exports[`<PetPage /> PetPage renders the PetCard with valid petData and Snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f6f6f6",
      },
      {
        "flex": 1,
        "justifyContent": "flex-start",
        "width": "100%",
      },
    ]
  }
  testID="pet-view"
>
  <View
    style={
      [
        {
          "backgroundColor": "#f6f6f6",
        },
        {
          "alignItems": "center",
        },
      ]
    }
    testID="pet-view"
  >
    <View
      style={
        [
          {
            "backgroundColor": "#f6f6f6",
          },
          {
            "alignItems": "center",
          },
        ]
      }
      testID="pet-view"
    >
      <Image
        resizeMode="cover"
        source={
          {
            "uri": "https://fakeurl/fakeimage/test.png",
          }
        }
        style={
          {
            "borderRadius": 9999,
            "height": 80,
            "width": 80,
          }
        }
        testID="pet-image"
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#155e75",
          },
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "lineHeight": 28,
          },
          {
            "paddingTop": 8,
          },
        ]
      }
      testID="pet-title"
    >
      info_pet_details
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      info_pet_name
      : 
      testPetName
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      info_pet_type
      : 
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      info_pet_gender
      : 
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      info_pet_breed
      : 
      testPetBreed
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      info_pet_age
      : 
      testPetAge
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      info_pet_medical_condition
      : 
      testPetMC
    </Text>
    <Text
      style={
        [
          {
            "color": "#155e75",
          },
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "lineHeight": 28,
          },
          {
            "paddingTop": 8,
          },
        ]
      }
      testID="pet-title"
    >
      info_owner_contact
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      info_owner_name
      : 
      testProfileEmail
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      testID="pet-text"
    >
      info_owner_phone: testProfileCountryCode testProfilePhone
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#075E54",
          "borderRadius": 10,
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
          "margin": 6,
          "opacity": 1,
          "padding": 8,
          "width": "85%",
        }
      }
      testID="pet-button"
    >
      <Text />
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#ffffff",
            },
          ]
        }
        testID="pet-text"
      >
        info_button_owner_whatsapp
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0e7490",
          "borderRadius": 10,
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
          "margin": 6,
          "opacity": 1,
          "padding": 8,
          "width": "85%",
        }
      }
      testID="pet-button"
    >
      <Text />
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#ffffff",
            },
          ]
        }
        testID="pet-text"
      >
        QR Code
      </Text>
    </View>
  </View>
</View>
`;
